docs/content/guide/component-router.ngdoc

```diff
@@ -855,7 +855,7 @@ function HeroDetailComponent(heroService) {
   ...
   this.gotoHeroes = function() {
     var heroId = this.hero && this.hero.id;
-    this.$router.navigate(['HeroList']);
+    this.$router.navigate(['HeroList', {id: heroId}]);
```

Comment:
I agree, the better solution would be to remove the `var heroId = ...` line. I try to change my pull-request.